chat-plugins/info.js

```diff
@@ -2614,18 +2616,35 @@ var commands = exports.commands = {
 		if (!this.can('declare', null, room)) return false;
 		if (!this.canBroadcast()) return;
 		if (this.room.isPersonal && !this.user.can('announce')) {
-			this.errorReply("Images are not allowed in personal rooms.");
-			return false;
+			return this.errorReply("Images are not allowed in personal rooms.");
 		}
 
 		var targets = target.split(',');
-		if (targets.length !== 3) {
-			return this.parse('/help showimage');
-		}
+		if (targets.length !== 3) return this.parse('/help showimage');
+
+		var image = targets[0].trim();
+		if (!image) return this.errorReply('No image URL was provided!');
+
+		var imageUrl = url.parse(image);
+		image = imageUrl.protocol || '//' + imageUrl.pathname;
+
+		var width = targets[1].trim();
+		if (!width) return this.errorReply('No width for the image was provided!');
+		if (!isNaN(width)) width += 'px';
+
+		var height = targets[2].trim();
+		if (!height) return this.errorReply('No height for the image was provided!');
+		if (!isNaN(height)) height += 'px';
+
+		width = Tools.escapeHTML(width);
+		height = Tools.escapeHTML(height);
 
-		this.sendReply('|raw|<img src="' + Tools.escapeHTML(targets[0]) + '" alt="" width="' + toId(targets[1]) + '" height="' + toId(targets[2]) + '" />');
+		this.sendReply('|raw|<img src="' + image + '" style="width: ' + width + '; height: ' + height + '" />');
```

Comment:
Not like Node's `url` library guarantees that the URL is valid in the first place, as far as I can tell.